lib/index.js

```diff
@@ -94,7 +94,33 @@ Insight.prototype.track = function () {
 	}).join('/');
 
 	// timestamp isn't unique enough since it can end up with duplicate entries
-	this._queue[Date.now() + ' ' + path] = path;
+	this._queue[Date.now() + ' ' + path] = {
+		path: path,
+		type: 'pageview'
+	};
+	this._save();
+};
+
+Insight.prototype.trackEvent = function (category, action, label, value) {
+	if (this.optOut) {
+		return;
+	}
+
+	if (this.trackingProvider !== 'google') {
+		throw new Error('event tracking is supported only for google-analytics');
+	}
+	if (!category || !action) {
+		throw new Error('event category and event action required');
+	}
+
+	// timestamp isn't unique enough since it can end up with duplicate entries
+	this._queue[Date.now() + ' ' + category + action + label + value] = {
```

Comment:
This will stringify undefined value, do we really want that?

Also, should we use a UUID instead?